silk-core/src/main/java/net/silkmc/silk/core/mixin/player/MixinInventory.java

```diff
@@ -0,0 +1,31 @@
+package net.silkmc.silk.core.mixin.player;
+
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.silkmc.silk.core.event.player.PlayerItemEvents;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(Inventory.class)
+public class MixinInventory {
+
+    @Shadow
+    @Final
+    public Player player;
+
+    @Inject(method = "add(Lnet/minecraft/world/item/ItemStack;)Z", at = @At(value = "HEAD"), cancellable = true)
```

Comment:
this is not the correct function for collecting items, this is just the inventory add function

Reply:
it's the only way to cancel the collecting

Reply:
hm?

Reply:
take a look by yourself. if you want the player to not collect the item you have to use this function